lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1914,270 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
+
+    try:
+        keyvault_poller = keyvault_client.vaults.delete(resource_group_name, vault_name)
+        log.info(f"{keyvault_poller}")
+        log.info(f"Key Vault {vault_name} deleted successfully.")
+    except Exception as e:
+        log.error(f"Error deleting Key Vault: {e}")
+        raise LisaException(str(e))
+
+
+def get_certificate_client(
+    vault_url: str, platform: "AzurePlatform"
+) -> CertificateClient:
+    return CertificateClient(vault_url, platform.credential)
+
+
+def get_secret_client(vault_url: str, platform: "AzurePlatform") -> SecretClient:
+    return SecretClient(vault_url, platform.credential)
+
+
+def create_certificate(
+    platform: "AzurePlatform",
+    vault_url: str,
+    log: Logger,
+    cert_name: str,
+    retries: int = 5,
+    delay: int = 2,
+) -> str:
+    certificate_client = get_certificate_client(vault_url, platform)
+    secret_client = get_secret_client(vault_url, platform)
+
+    cert_policy = CertificatePolicy.get_default()
+
+    for attempt in range(retries):
+        try:
+            # Create certificate
+            create_certificate_result = certificate_client.begin_create_certificate(
+                cert_name, policy=cert_policy
+            )
+            log.info(
+                f"Certificate '{cert_name}' has been created. "
+                f"Result: {create_certificate_result}"
+            )
+            certificate_client.update_certificate_properties(
+                certificate_name=cert_name, enabled=True
+            )
+            break
+        except Exception as e:
+            if attempt < retries - 1:
+                sleep(delay)
+            else:
+                log.error(
+                    f"Failed to create certificate named '{cert_name}' "
+                    f"after {retries} retries. Error: {e}"
+                )
+                raise LisaException(str(e))
+
+    try:
+        secret_id: Optional[str] = secret_client.get_secret(name=cert_name).id
+        if secret_id:
+            secret_url_without_version = secret_id.rsplit("/", 1)[0]
+            return secret_url_without_version
+        else:
+            log.error(f"Failed to retrieve secret ID:'{cert_name}'.")
+            raise LisaException(f"Failed to retrieve secret ID:'{cert_name}'.")
+    except Exception as e:
+        log.error(f"Failed to retrieve secret ID:'{cert_name}': {e}")
+        raise LisaException(str(e))
+
+
+def check_certificate_existence(
+    vault_url: str, cert_name: str, log: Logger, platform: "AzurePlatform"
+) -> bool:
+    certificate_client = CertificateClient(
+        vault_url=vault_url, credential=platform.credential
+    )
+    try:
+        # Try to get the certificate. If this call succeeds, the certificate exists.
+        certificate_client.get_certificate(cert_name)
+        return True
+    except Exception as e:
+        log.error(f"Failed check existance of cert:'{cert_name}': {e}")
```

Comment:
Don't log error, the error log is hard to check. Raise exception directly. If it doesn't need to fail a case, catch it in test case level.